data_provenance/to-dolma.py

```diff
@@ -66,6 +66,22 @@
 SOURCE_NAME = "Data Provenance Initiative"
 
 
+def get_unique_ids(folder_path):
+    unique_doc_ids = set()
+
+    jsonl_files = [f for f in os.listdir(folder_path) if f.endswith(".jsonl.gz")]
+
+    for filename in tqdm(jsonl_files, desc="Processing files"):
+        file_path = os.path.join(folder_path, filename)
+        with gzip.open(file_path, "rt") as f:
+            for line in f:
+                data = json.loads(line)
+                doc_id = data.get("id")
```

Comment:
Each dolma example from this pipeline will have a `.["meta"]["dataset_id"]` key. Can we use that instead of munging the example id?

Reply:
Rerunning this now to make sure looks alright

Reply:
Reran and works. got 315